spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskTemplate.java

```diff
@@ -145,14 +146,43 @@ public class TaskTemplate implements TaskOperations {
 
 		this.restTemplate = restTemplate;
 		this.dataFlowServerVersion = dataFlowServerVersion;
+		String version = VersionUtils.getThreePartVersion(dataFlowServerVersion);
+		if (VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, EXECUTIONS_CURRENT_RELATION_VERSION)) {
+			Assert.isTrue(resources.getLink(EXECUTIONS_CURRENT_RELATION).isPresent(), ()-> EXECUTIONS_CURRENT_RELATION + " relation is required");
+			this.executionsCurrentLink = resources.getLink(EXECUTIONS_CURRENT_RELATION).get();
+		} else {
+			this.executionsCurrentLink = null;
+		}
+		if (VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, VALIDATION_RELATION_VERSION)) {
+			Assert.isTrue(resources.getLink(VALIDATION_REL).isPresent(), ()->VALIDATION_REL + " relation is required");
+			this.validationLink = resources.getLink(VALIDATION_REL).get();
+		} else {
+			this.validationLink = null;
+		}
 
-		this.validationLink = resources.getLink(VALIDATION_REL).get();
-		this.aboutLink = resources.getLink("about").get();
 
-		this.thinExecutionsLink = resources.getLink(THIN_EXECUTIONS_RELATION).orElse(null);
-		this.thinExecutionsByNameLink = resources.getLink(THIN_EXECUTIONS_BY_NAME_RELATION).orElse(null);
-		this.executionLaunchLink = resources.getLink(EXECUTION_LAUNCH_RELATION).orElse(null);
-		this.executionsCurrentLink = resources.getLink(EXECUTIONS_CURRENT_RELATION).orElse(null);
+		if(VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(version, VALIDATION_THIN_TASK_VERSION) && resources.getLink(THIN_EXECUTIONS_RELATION).isPresent()) {
```

Comment:
I would consider moving this logic into a well-named helper method to avoid the repeat checks as its hard to read currently.

Reply:
Removed the `resources.getLink(THIN_EXECUTIONS_RELATION).isPresent()` from if